Src/FluentAssertions/CallerIdentifier.cs

```diff
@@ -69,6 +70,16 @@ private static string ExtractVariableNameFrom(StackFrame frame)
             return caller;
         }
 
+        private static bool UsesNewKeyword(string candidate)
+        {
+            return new Regex(@"new(\s?\[|\s?\{|\s\w+)").IsMatch(candidate);
```

Comment:
Just nitpicking, but the static [`Regex.IsMatch`](https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.ismatch?view=netframework-4.7.1#System_Text_RegularExpressions_Regex_IsMatch_System_String_System_String_) caches the regex.

Reply:
Good point. I still have to clean up the code. Currently focusing on getting all tests green.